term.go

```diff
@@ -497,3 +504,20 @@ func (t *Terminal) Dragged(d *fyne.DragEvent) {
 func (t *Terminal) DragEnd() {
 	t.selecting = false
 }
+
+// SetMiddleware sets the middleware function that will be used when creating a new terminal.
+// The middleware function is responsible for setting up the I/O readers and writers.
```

Comment:
The way middleware is implemented it is an interface type not function. 
It feels like a function would be enough to set up the interception?

Reply:
thats the beauty of golang right?
```
// MiddlewareFunc is a function type that matches the signature of the
// SetupMiddleware method in the Middleware interface.
type MiddlewareFunc func(r io.Reader, w io.WriteCloser) (io.Reader, io.WriteCloser)

// SetupMiddleware allows MiddlewareFunc to satisfy the Middleware interface.
// It calls the MiddlewareFunc itself.
func (m MiddlewareFunc) SetupMiddleware(r io.Reader, w io.WriteCloser) (io.Reader, io.WriteCloser) {
	return m(r, w)
}
```

Reply:
I guess if following the http middleware / handler idea it was missing the func helper? https://pkg.go.dev/net/http#HandlerFunc

Reply:
yes essentially the same pattern